src/java.base/share/classes/sun/net/www/protocol/file/FileURLConnection.java

```diff
@@ -71,20 +69,17 @@ protected FileURLConnection(URL u, File file) {
      */
     public void connect() throws IOException {
         if (!connected) {
-            try {
-                filename = file.toString();
-                isDirectory = file.isDirectory();
-                if (isDirectory) {
-                    String[] fileList = file.list();
-                    if (fileList == null)
-                        throw new FileNotFoundException(filename + " exists, but is not accessible");
-                    files = Arrays.<String>asList(fileList);
-                } else {
-                    is = new BufferedInputStream(new FileInputStream(filename));
-                }
-            } catch (IOException e) {
-                throw e;
+
+            isDirectory = file.isDirectory();
+            if (isDirectory) {
+                String[] fileList = file.list();
+                if (fileList == null)
+                    throw new FileNotFoundException(file.getPath() + " exists, but is not accessible");
+                directoryListing = Arrays.<String>asList(fileList);
```

Comment:
Is the `<String>` parameterization necessary? I tested on jshell, and it seems since the destination variable's type is `List<String>` we don't need this explicit parameterization to distinguish varargs.

Reply:
Yes, indeed this can be dropped. See 06b2a23.